packages/terra-functional-testing/tests/jest/services/wdio-docker-service.test.js

```diff
@@ -0,0 +1,302 @@
+
+      expect(service.removeStack).toHaveBeenCalled();
+    });
+
+    it('should wait for the network to become ready', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'removeStack').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkReady').mockImplementationOnce(() => Promise.resolve());
+
+      await service.deployStack();
+
+      expect(service.awaitNetworkReady).toHaveBeenCalled();
+    });
+  });
+
+  describe('removeStack', () => {
+    it('should not execute the docker stack removal if it is already removed', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: '' }));
+
+      await service.removeStack();
+
+      expect(mockExec).toHaveBeenCalledTimes(1);
+      expect(mockExec).toHaveBeenCalledWith('docker stack ls | grep wdio || true');
+    });
+
+    it('should remove the docker stack', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: 'Active' }));
+
+      jest.spyOn(service, 'awaitServiceRemoval').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkRemoval').mockImplementationOnce(() => Promise.resolve());
+
+      await service.removeStack();
+
+      expect(mockExec).toHaveBeenCalledTimes(2);
+      expect(mockExec).toHaveBeenCalledWith('docker stack rm wdio');
+    });
+
+    it('should wait for the network and service to be removed', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: 'Active' }));
+
+      jest.spyOn(service, 'awaitServiceRemoval').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkRemoval').mockImplementationOnce(() => Promise.resolve());
+
+      await service.removeStack();
+
+      expect(service.awaitServiceRemoval).toHaveBeenCalled();
+      expect(service.awaitNetworkRemoval).toHaveBeenCalled();
+    });
+  });
+
+  describe('pollCommand', () => {
+    it('should invoke the command and callback', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve('mock output'));
+
+      const mockCallback = jest.fn().mockImplementation(() => Promise.resolve());
+
+      await service.pollCommand('mock command', mockCallback);
+
+      expect(mockExec).toHaveBeenCalledWith('mock command');
+      expect(mockCallback).toHaveBeenCalledWith('mock output');
+    });
+
+    it('should timeout if the retry count is exceeded', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve());
+
+      const mockCallback = jest.fn().mockImplementation(() => Promise.reject());
+
+      try {
+        // Retry 3 times. Repeat every 10 milliseconds.
+        await service.pollCommand('mock command', mockCallback, 3, 10);
```

Comment:
is the the only place we expect the retry count & poll time to be used? I didn't notice it being used in the class

Reply:
Yes, unfortunately I ended up adding it to simplify testing. Without these overrides the test to reach the timeout will take 60 seconds. The poll function is intended to be re-usable, so the additional parameters don't seem harmful and could be beneficial/utilized with configuration options in the future.